File: 1525-number-of-good-ways-to-split-a-string/1525-number-of-good-ways-to-split-a-string.java
class Solution {
    public int numSplits(String s) {
       int n=s.length();
        Map<Character,Integer>rightmap=new HashMap<>();
        for(int i=0;i<n;i++) rightmap.put(s.charAt(i),rightmap.getOrDefault(s.charAt(i),0)+1);
        Map<Character,Integer>leftmap=new HashMap<>();
        int cnt=0;
        for(int i=0;i<n;i++){
            rightmap.put(s.charAt(i),rightmap.get(s.charAt(i))-1);
            if(rightmap.get(s.charAt(i))==0) rightmap.remove(s.charAt(i));
            leftmap.put(s.charAt(i),leftmap.getOrDefault(s.charAt(i),0)+1);
            if(leftmap.size()==rightmap.size()) cnt++;
        }
        return cnt;
    }
}

File: 1525-number-of-good-ways-to-split-a-string/1525-number-of-good-ways-to-split-a-string.cpp
class Solution {
public:
    int numSplits(string s) {
        int n=s.length();
          map<char,int>rightmap;
        for(int i=0;i<n;i++) rightmap[s[i]]++;
        map<char,int>leftmap;
        int cnt=0;
        for(int i=0;i<n;i++){
            rightmap[s[i]]--;
            if(rightmap[s[i]]==0) rightmap.erase(s[i]);
            leftmap[s[i]]++;
            if(leftmap.size()==rightmap.size()) cnt++;
        }
        return cnt;
    }
};